apis/coralogix/v1alpha1/alert_types.go

```diff
@@ -1197,20 +1197,59 @@ func DeepEqualNotificationGroups(notificationGroups []NotificationGroup, actualN
 		}
 	}
 
-	for i := range notificationGroups {
-		notificationGroup, actualNotificationGroup := notificationGroups[i], actualNotificationGroups[i]
-		{
-			if equal, diff := notificationGroup.DeepEqual(actualNotificationGroup); !equal {
-				return false, utils.Diff{
-					Name:    fmt.Sprintf("Notifications.%d.%s", i, diff.Name),
-					Desired: diff.Desired,
-					Actual:  diff.Actual,
+	desiredGroupByFieldsToNotification := getGroupByFieldsToNotificationMap(notificationGroups)
+	actualGroupByFieldsToNotification := getGroupByFieldsToNotificationMap(actualNotificationGroups)
+	for groupByFields, notifications := range desiredGroupByFieldsToNotification {
+		if actualNotifications, ok := actualGroupByFieldsToNotification[groupByFields]; !ok {
+			return false, utils.Diff{
+				Name:    fmt.Sprintf("Notifications.gropup-by:%s", groupByFields),
+				Desired: notifications,
+				Actual:  nil,
+			}
+		} else {
```

Comment:
This else statement could be removed to reduce one indentation level, making the code easier to read :)

https://dev.to/zachsnoek/clean-up-your-code-by-removing-unneeded-indentation-3g9i

Reply:
But then I'll need to declare on actualNotifications and ok outside the if-else (line 1203).
What do you prefer?

Reply:
The if remains there, just the else that goes away:

```go
		if actualNotifications, ok := actualGroupByFieldsToNotification[groupByFields]; !ok {
			return false, utils.Diff{
				Name:    fmt.Sprintf("Notifications.gropup-by:%s", groupByFields),
				Desired: notifications,
				Actual:  nil,
			}
		} 

		desiredNotificationsByIntegrationName := getNotificationsByIntegrationNameMap(notifications)
		actualNotificationsByIntegrationName := getNotificationsByIntegrationNameMap(actualNotifications)

		for integrationName, notification := range desiredNotificationsByIntegrationName {
			if actualNotification, ok := actualNotificationsByIntegrationName[integrationName]; !ok {
				return false, utils.Diff{
					Name:    fmt.Sprintf("Notifications.gropup-by:%s.%s", groupByFields, integrationName),
					Desired: notification,
					Actual:  nil,
				}
			} else if equal, diff := notification.DeepEqual(actualNotification); !equal {
				return false, utils.Diff{
					Name:    fmt.Sprintf("Notifications.gropup-by:%s.%s.%s", groupByFields, integrationName, diff.Name),
					Desired: diff.Desired,
					Actual:  diff.Actual,
				}
			}
		}
```